Makefile

```diff
@@ -207,15 +207,50 @@ gfx/pokemon/girafarig/front.animated.tilemap: gfx/pokemon/girafarig/front.2bpp g
 	tools/pokemon_animation_graphics --girafarig -t $@ $^
 
 
-### Misc file-specific graphics rules
+### Pokemon and trainer sprite rules
+
+gfx/pokemon/%/back.2bpp: rgbgfx += --columns
+gfx/pokemon/%/back.2bpp: gfx/pokemon/%/back.png gfx/pokemon/%/normal.gbcpal
+	$(RGBGFX) $(rgbgfx) --colors gbc:$(word 2,$^) -o $@ $<
+gfx/pokemon/%/front.2bpp: gfx/pokemon/%/front.png gfx/pokemon/%/normal.gbcpal
+	$(RGBGFX) $(rgbgfx) --colors gbc:$(word 2,$^) -o $@ $<
+gfx/pokemon/%/normal.gbcpal: gfx/pokemon/%/front.gbcpal gfx/pokemon/%/back.gbcpal
+	tools/gbcpal $(tools/gbcpal) $@ $^
+
+gfx/trainers/%.2bpp: rgbgfx += --columns
+gfx/trainers/%.2bpp: gfx/trainers/%.png gfx/trainers/%.gbcpal
+	$(RGBGFX) $(rgbgfx) --colors gbc:$(word 2,$^) -o $@ $<
+
+# Egg does not have a back sprite, so it only uses front.gbcpal
+gfx/pokemon/egg/front.2bpp: gfx/pokemon/egg/front.png gfx/pokemon/egg/front.gbcpal
+gfx/pokemon/egg/front.2bpp: rgbgfx += --colors gbc:$(word 2,$^)
```

Comment:
you sure this last line is necessary? I think the flag will just be specified twice now.

Reply:
Nope, this uses the generic 2bpp-from-png rule below. If `egg` used the `gfx/pokemon/%/front.2bpp` rule above, this whole line would be unnecessary.

(Why doesn't it use the pattern rule? Because the rule's dependencies are front.png, which exists, and normal.gbcpal, which does not; following the dependency chain for making normal.gbcpal leads to needing back.png, which does not exist and has no rule to make one, so the pattern doesn't apply.)